### ISSUE TITLE
[Product] Shop Pay logo missing accessible name

### ISSUE THREAD

**Issue by nicklepine** [Shopify/dawn#75]
## Issue summary

When reading product content using a screen reader, the Shop Pay logo image is missing its accessible name/alternative text. This may create some confusion as the installment content would seem to be incomplete.

<details>
  <summary>Screenshots</summary>
<img width="597" alt="Product page, Shop Pay logo is highlighted." src="https://user-images.githubusercontent.com/1392632/121944831-1c755b00-cd21-11eb-83de-ecacf7b556da.png">

</details>

#### HTML

```html
<shop-pay-logo
  role="img"
  aria-labelledby="shopify-payment-terms-shop-pay-logo"
  …
></shop-pay-logo>
```

## Steps to reproduce

1. Go to a product page (Using Dawn) with any supported browser
2. With an active screen reader, use the virtual cursor to navigate to the Shop Pay logo.

### Expected

- Screen reader to announce image as, _"Shop Pay, image"_.

### Actual

- Image is skipped over.

## Recommendation

Adjust the markup of the image. Include the alternative text inline via `aria-label`, as apposed to referencing another node via `aria-labelledby`. I believe this to be the issue; the referenced node exists within a shadow-dom component which is not reachable.

#### HTML

```html
<shop-pay-logo
  role="img"
  aria-label="Shop Pay"
  …
></shop-pay-logo>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Component:** Global
- **WCAG Principal:** Perceivable
- **WCAG SC:** [1.1.1 Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
- **Severity:** Medium
- **Effort:** Low

**Comment by ludoboludo** [comment]
Hmm I don't think this would be tackled by us. As this is something coming from the platform 🤔  
We're just outputting: `{{ form | payment_terms }}`

I will try and see who could help us with this.
